toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -300,6 +301,49 @@ func (g *PkgGraph) validateNodeForLookup(pkgNode *PkgNode) (valid bool, err erro
 	return
 }
 
+/*check if a runNode already exists for the same version of the remoteNode, if exists, replace runNode in the lookup with remoteNode*/
+func (g *PkgGraph) AddRemoteToLookup(pkgNode *PkgNode, deferSort bool) (err error) {
```

Comment:
If we want this to be an external API, I don't think we would ever expect deferSort to be true. The graph should always be established at this point and every addition needs to be sorted right away.

Reply:
```suggestion
func (g *PkgGraph) AddRemoteToLookup(pkgNode *PkgNode) (err error) {
```